### PR TITLE
[tilelive-* release required] Use different buffer-size values for different tile formats

### LINKED ISSUE #548: Use different buffer-size values for different tile formats
We're going to need a way to handle different `buffer-size` values for different formats. For example, [this issue](https://github.com/CartoDB/Windshaft/issues/522) requests to have a default `buffer-size` of 0 for MVT format.

####  Tasks
- [ ] Add `buffer-size` property with different values for different formats in the Mapconfig. For example:
   ```
   buffersize: {
       png: 128,
       grid: 64,
       mvt: 0
   }
   ```
- [ ] Update Mapconfig documentation and release new version with the added changes

### PR DESCRIPTION
**Pull request by dgaubert**
Fixes #548

### CODE REVIEW

File: doc/MapConfig-1.6.0.md [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+    // default: undefined (infinite)
+    maxzoom: 18,
+
+    // OPTIONAL
+    // minzoom to be renderer. From this zoom tiles will respond 404. Must be less than maxzoom
+    // default: 0
+    minzoom: 3,
+
+    // OPTIONAL
+    // Extra tolerance around the map (in pixels) used to ensure labels crossing tile boundaries are equally rendered
```
Comment (rochoa) [inline doc/MapConfig-1.6.0.md]: I'm missing the documentation related to what priority this has. Does this supersede the buffer-size in the CartoCSS or the other way around?

File: doc/MapConfig-1.6.0.md [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+    // minzoom to be renderer. From this zoom tiles will respond 404. Must be less than maxzoom
+    // default: 0
+    minzoom: 3,
+
+    // OPTIONAL
+    // Extra tolerance around the map (in pixels) used to ensure labels crossing tile boundaries are equally rendered 
+    // in each tile (e.g. cut in each tile).
+    // 
+    // To configure `buffer-size` for any tile format: 
+    //  - buffersize: 128
```
Comment (rochoa) [inline doc/MapConfig-1.6.0.md]: What about simplifying this? Not a big fan of having different ways/types to set a configuration.

Comment (ethervoid) [inline doc/MapConfig-1.6.0.md]: We're doing the same way `metatile` property is doing but we could just let the buffer-size definition by format. What do you think?

Comment (rochoa) [inline doc/MapConfig-1.6.0.md]: I see 😄 . But with the metatile we already had the one with just one metatile for all formats and we added the _by format_ one.

So let's go with your proposal, just expose the _by format_ one, ok?

File: doc/MapConfig-1.6.0.md [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+    // To configure `buffer-size` per tile format:
+    //  - buffersize: {
+    //      'png': 64,
+    //      'grid.json': 64,
+    //      'mvt': 0
+    //    }
+    //
+    // Buffer-size also can be configured through `cartocss`, in these cases the priority to apply the right `buffer-size` is:
+    //  - Raster: cartocss >> map-config >> default
+    //  - Vector: map-config >> default
```
Comment (rochoa) [inline doc/MapConfig-1.6.0.md]: What happens if I define `Map { buffer-size: 64 }` in one of my layers' CartoCSS and `buffersize: {mvt: 0}` in here, and I retrieve a MVT tile? Do we ignore the buffer size defined through the CartoCSS property in that scenario?

Comment (ethervoid) [inline doc/MapConfig-1.6.0.md]: Yes, we don't take into account the `buffer-size` value defined in CSS for the `mvt` tiles

Comment (dgaubert) [inline doc/MapConfig-1.6.0.md]: Please note that it's the default behaviour in `tilelive-bridge`. For vector formats they create a map from the `xml` and then always override the `buffer-size` value, before our changes always to 256 and now with a custom value. On the contrary, `tilelive-mapnik` has another approach creating the map instance with a default value of `buffer-size` and then it's overridden. As we want to sent a PR to upstream for `tilelive-bridge` we didn't take into account to change that behaviour.

**Pull request merged**